### ISSUE TITLE
Support CSP `require-trusted-types-for`

### ISSUE THREAD

**Issue by xamir82** [sveltejs/kit#7975]
### Describe the problem

Even though `require-trusted-types-for` seems to be present in the list of CSP directives in the configuration, setting it (to `script`, which is its only value) will actually cause SvelteKit to fail to function properly:

![image](https://user-images.githubusercontent.com/119128472/206075347-9149652a-c0d7-4f7e-8fa4-a56ab1515c67.png)

### Describe the proposed solution

I don't know much about this directive but it seems like what is assigned to `innerHTML` properties and whatnot, must be a special "trusted" type — see [this article](https://web.dev/trusted-types/).

SvelteKit (or probably Svelte, more specifically) does do a lot of these things (e.g. changing `innerHTML` properties and so on), so SvelteKit should probably account for this when this directive is enabled.

### Importance

nice to have

**Comment by AlanBreck** [comment]
Can you confirm that these errors are due to Svelte and not your own code? In my case, the errors were caused by setting `.src` on a script tag in the site analytics set up.

**Comment by fallaciousreasoning** [comment]
Hey so I've got a PR for Svelte that will fix this for a very specific subset of this problem (CSS for custom elements) https://github.com/sveltejs/svelte/pull/8135

You could probably do something similar to fix the above issue, as looking at the console output it looks like this only applies to updating styles (which we can do via `textContent` rather than `innerHTML` which won't trigger the `TrustedTypes` check).

From a higher level, it would probably be a good idea to add a few fine grained trusted types policies which Svelte would then use when doing something that could trigger a violation. That way consumers of svelte could whitelist the policies in their CSP.

**Comment by aradalvand** [comment]
I get the same error:
![image](https://github.com/sveltejs/kit/assets/26527405/b00eb6e1-4082-42af-9794-297b43b09605)

**Comment by eltigerchino** [comment]
This silly snippet makes the error go away
```svelte
<script>
	onMount(() => {
		if (window.trustedTypes && trustedTypes.createPolicy) {
			trustedTypes.createPolicy('default', {
				createHTML: (str, sink) => str
			});
		}
	})
</script>
```

**Comment by eltigerchino** [comment]
Agreed. I've filed an issue for this in the Svelte repo https://github.com/sveltejs/svelte/issues/14438

**Comment by kiwixz** [comment]
I believe this is fixed now? The Svelte issue was just closed.

**Comment by teemingc** [comment]
@Rich-Harris are there changes we need to make to kit in lieu of https://github.com/sveltejs/svelte/pull/16271 ?

EDIT: I think so. We should identify the csp config options then add the header. It will require the newer svelte version

**Comment by teemingc** [comment]
Ok so this already works if folks update their Svelte version to >= 5.51.0.
You should add 'svelte-trusted-html' to `trusted-types`.

**Comment by kiwixz** [comment]
I just tried it; I think there may still be a problem with the service worker registration.

> TypeError: Failed to execute 'register' on 'ServiceWorkerContainer': This document requires 'TrustedScriptURL' assignment.

**Comment by teemingc** [comment]
Thanks for trying it out. Looks like we need to add https://developer.mozilla.org/en-US/docs/Web/API/TrustedScriptURL during the service worker registration
